### DISCUSSION TITLE
How to capture vehicle in the Images

### DISCUSSION THREAD

**Discussion by UrwinG** [microsoft/AirSim#4556] [Support Q&A]
Hi there, I am hoping to generate data of my vehicles and would like them to be in the scene. By default I noticed the images are captured from the POV of the vehicle and I was wondering how I can change that to have the vehicle in the frame as well

Thanks!

**Comment by alonfaraj** [comment]
Hi @UrwinG,
You can add a custom camera and set its position behind your vehicle.
Somthing like this:
<details>
<summary>settings</summary>

```
{
  "SettingsVersion": 1.2,
  "SimMode": "Car",
  "EnableRpc": true,
  "ViewMode": "Fpv",
  "LogMessagesVisible": true,
  "EngineSound": false,

  "Vehicles": {
    "PhysXCar": {
      "VehicleType": "PhysXCar",
      "DefaultVehicleState": "",
      "AutoCreate": true,
      "PawnPath": "",
      "EnableCollisionPassthrogh": false,
      "EnableCollisions": true,
      "RC": {
        "RemoteControlID": 0
      },
      			
		"Cameras": {
			"cam_0": {
				"CaptureSettings": [
					{
						"ImageType": 0,
						"Width": 1280,
						"Height": 720,
						"FOV_Degrees": 90.0,
						"AutoExposureSpeed": 100,
						"AutoExposureBias": 0,
						"AutoExposureMaxBrightness": 0.64,
						"AutoExposureMinBrightness": 0.03,
						"MotionBlurAmount": 0,
						"TargetGamma": 1.0,
						"ProjectionMode": "",
						"OrthoWidth": 5.12
					}
				],
				"X": -6, "Y": 0, "Z": -3,
				"Pitch": -7.0, "Roll": 0.0, "Yaw": 0.0
			}
		}
    }
},
	"SubWindows": [
		{"WindowID": 0, "CameraName": "cam_0", "ImageType": 0, "VehicleName": "", "Visible": true, "External": false}
	]
}
```

</details>

**Reply by UrwinG** [reply]
Thank you so much, going to give this a go right now!

I got another question if you don't mind. Is there a way to use the WeatherActor blueprint to be used when not having the AirSimGamemode on? Just if I want to use the neat weather effects in other levels/ projects

**Reply by UrwinG** [reply]
Just tried the code you gave me and I do see the one subwindow displays the car from the back. How can I actually output that camera and have the images in a folder somewhere. It currently just does the main viewport still which is the first person view of the driver

**Reply by alonfaraj** [reply]
> Just tried the code you gave me and I do see the one subwindow displays the car from the back. How can I actually output that camera and have the images in a folder somewhere. It currently just does the main viewport still which is the first person view of the driver

You can set "recording" in the settings file to record your custom camera. Take a look at the documentation to see an example. 
Another approach is to use the python API - simGetImage with your camera name.

**Reply by alonfaraj** [reply]
> Thank you so much, going to give this a go right now!
> 
> 
> 
> I got another question if you don't mind. Is there a way to use the WeatherActor blueprint to be used when not having the AirSimGamemode on? Just if I want to use the neat weather effects in other levels/ projects

I really don't know. I will have to take a deeper look. 
Or you can just try to migrate this actor to a clean project without airsim and check if you can use it out of the box.

**Reply by UrwinG** [reply]
Ill take a look at the record settings thanks!

Do you think it would be better to do "ComputerVision" mode and then have that camera look at my targets?

**Reply by UrwinG** [reply]
Yeah I tried just throwing in the WeatherActor and had no luck. The Gamemode had to be AirSimeGameMode

**Reply by alonfaraj** [reply]
> Ill take a look at the record settings thanks!
> 
> 
> 
> Do you think it would be better to do "ComputerVision" mode and then have that camera look at my targets?

Depends on your use case.  
If your targets are non airsim actors and you don't need the airsim car to use other sensors which are not supported in computer vision mode, so yes. 
CV mode will be much easier to navigate in the scene and take photos of your target from any angle.

**Reply by UrwinG** [reply]
yeah this is exactly what I want!!

So in simple terms I have a new blueprint pawn created (in this case its CoastGuard)
All i really want to do is spawn in the camera that captures the information and images and just put it on an animation track to film the CoastGuard

So right now in my settings I have: 

{
  "SettingsVersion": 1.2,
  "CameraDefaults": {
    "CaptureSettings": [
      {
        "ImageType": 0,
        "Width": 1920,
        "Height": 1080,
        "FOV_Degrees": 90,
        "AutoExposureSpeed": 100,
        "MotionBlurAmount": 0
      }
    ]
  },
  "SimMode": "ComputerVision",

  "DefaultComputerVision": {
    "PawnBP": "Blueprint'/Game/AirSim/Blueprints/CoastGuard.CoastGuard"
  },



    "PawnPaths": {
      "CoastGuard": { "PawnBP": "Blueprint'/Game/AirSim/Blueprints/CoastGuard.CoastGuard" }
    },
    "Vehicles": {
      "MyVehicle": {
        "VehicleType": "ComputerVision",
        "PawnPath": "CoastGuard"
      }
    }
 
}



It says it fails to load my pawn but maybe I dont even need to do this?

Thanks again for all the help!!

**Reply by UrwinG** [reply]
Also, I have no AirSim related content in my scene but when I try to add the "BP_ComputerVisionPawn+ my project crashes everytime. I feel like if I can just add that to my scene I would almost be finished

**Reply by alonfaraj** [reply]
Do you have any reason to create a custom pawn instead the default airsim CV?

If you do, the template should be like this:

 "CustomPawn": {"PawnBP": "Class'/Game/Assets/Blueprints/MyPawn.MyPawn_C'"}

**Reply by UrwinG** [reply]
> Do you have any reason to create a custom pawn instead the default airsim CV?
> 
> If you do, the template should be like this:
> 
> "CustomPawn": {"PawnBP": "Class'/Game/Assets/Blueprints/MyPawn.MyPawn_C'"}



So to do this do I make a blueprint -> pawn and then write the above line of code in my settings? I tried doing that it kept saying it failed to load asset

**Reply by alonfaraj** [reply]
take a look at the explanation how to create your own pawn derived from airsim pawn.
https://microsoft.github.io/AirSim/settings/#pawnpaths

You need to follow the rules to make it work using airsim.
Is your pawn derived from CarPawn?
Did you spell it correctly in the settings?
